### DISCUSSION TITLE
Filter-like search in various columns of a treegrid

### DISCUSSION THREAD

**Discussion by vozvik** [mar10/wunderbaum#12] [Ideas]
Hi Martin,

I am trying to implement a search facility in a treegrid, similar to what is available as a filter already. The main point is that I would not only search in node.title but also in other columns.
(e.g. think of a grid of book related infos including a price column, and the user could enter a search term like price<30)

Is there a way to achieve this with a custom filter callback in the future?
For now I did a quick implementation that uses findAll() and it works reasonably well but the best I can do is select nodes with a match; I cannot hide unmatched nodes.
(the only quick idea I have as a workaround is "patching" titles of matching nodes with special characters and running a filter to only show those)

Thank you!

Cheers
Viktor

**Comment by mar10** [comment]
This is already implemented, but not yet exposed:
Try
```js
tree.extensions.filter.filterNodes(filter, options)
```
`filter` can be a string or a callback: 
https://mar10.github.io/wunderbaum/api/modules/types.html#NodeFilterCallback

`options` are optional: https://mar10.github.io/wunderbaum/api/interfaces/types.FilterNodesOptions.html

I will add it to the tree namespace soon.

**Reply by vozvik** [reply]
Exactly what I need, awesome! Thank you.
